### PR TITLE
[API] Finish covering taxation scenarios

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13 |
| Bug fix?        | kinda                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | -                      |
| License         | MIT                                                          |

### CODE REVIEW

File: src/Sylius/Bundle/TaxationBundle/Resources/config/doctrine/model/TaxRate.orm.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
         <field name="code" column="code" type="string" unique="true" />
         <field name="name" column="name" type="string" />
-        <field name="amount" column="amount" type="decimal" precision="10" scale="5" />
+        <field name="amount" column="amount" type="decimal" precision="10" scale="5" nullable="true" />
```
Comment (diimpp) [inline src/Sylius/Bundle/TaxationBundle/Resources/config/doctrine/model/TaxRate.orm.xml]: Why is this change necessary? Do TaxRate without amount exists?

Comment (NoResponseMate) [reply src/Sylius/Bundle/TaxationBundle/Resources/config/doctrine/model/TaxRate.orm.xml]: Yeah I went a bit too easy on this one. Reverted, and treating removing/null values for amount as 0, since that's what happens on the model.
Guessing the field being nullable is there just to satisfy passing an empty value through the form. Can't really change it there due to BC, but it would be nice to clean those in the future

File: features/taxation/managing_tax_rates/tax_rate_validation.feature
Comment (jakubtobiasz) [inline features/taxation/managing_tax_rates/tax_rate_validation.feature:1]: Why do we add these `@no-api` tags? On the first glance, they seem pretty easy to be implemented 🤔.

**Review by jakubtobiasz — APPROVED** [review]

**Review by Wojdylak — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, @NoResponseMate!
